### PR TITLE
Document http error codes

### PR DESCRIPTION
**Pull request by alexanderkjeldaas**
It's more useful to see some examples when we're trying to see which error to return.

### CODE REVIEW

**Comment by alpmestan** [comment]
Great initiative, examples rarely hurt :+1: LGTM

**Comment by dredozubov** [comment]
> throwErr $ err503 { errBody = "We're rewriting in PHP." }

:+1: 

but srsly, LGTM

File: servant-server/src/Servant/Server/Internal/ServantErr.hs [hunk 1]
```diff
@@ -163,6 +303,13 @@ err413 = ServantErr { errHTTPCode = 413
                     , errHeaders = []
                     }
 
+-- | 'err414' Request-URI Too Large
+--
+-- Example:
+--
+-- > failingHandler :: ExceptT ServantErr IO ()
+-- > failingHandler = throwErr $ err414 { errBody = "Maxiumu length is 64." }
```
Comment (3noch) [inline servant-server/src/Servant/Server/Internal/ServantErr.hs]: Oops!

Comment (alexanderkjeldaas) [inline servant-server/src/Servant/Server/Internal/ServantErr.hs]: Fixed. Thanks!

**Comment by soenkehahn** [comment]
LGTM, I'll merge.

**Pull request merged**